### PR TITLE
use current method signatures

### LINKED ISSUE #95937: es-image as part of fluent-elasticsearch cluster addon does not build
**What happened**:

es-image as part of fluent-elasticsearch cluster addon does not build

**What you expected to happen**:

`docker build .` in the es-image folder should work

**How to reproduce it (as minimally and precisely as possible)**:

`cd cluster/addons/fluentd-elasticsearch/es-image
make`

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by coffeepac**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

fixes discovery code in elasticsearch so it builds again

**Which issue(s) this PR fixes**:

Fixes #95937

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cluster/addons/fluentd-elasticsearch/es-image/elasticsearch_logging_discovery.go [hunk 1]
```diff
@@ -103,7 +102,7 @@ func main() {
 	namespace := metav1.NamespaceSystem
 	envNamespace := os.Getenv("NAMESPACE")
 	if envNamespace != "" {
-		if _, err := client.CoreV1().Namespaces().Get(context.TODO(), envNamespace, metav1.GetOptions{}); err != nil {
```
Comment (hasheddan) [inline cluster/addons/fluentd-elasticsearch/es-image/elasticsearch_logging_discovery.go:106]: I don't think these contexts can be removed.

Comment (coffeepac) [reply cluster/addons/fluentd-elasticsearch/es-image/elasticsearch_logging_discovery.go:106]: its confusing though, I'm unable to build this with the contexts present.  If I remove them, my IDE yells at me but both image build and image execution are successful (ie:  I get a functioning es cluster).

this is the source of the CI errors as well.

do you have any suggestions as to how I can dig into why I'm unable to build with the contexts present?  Is there a client go pinning issue I need to run down?

**Review by monotek — CHANGES_REQUESTED** [review]
any reason not going for the most current version 7.9.3?

**Comment by coffeepac** [comment]
@monotek limiting the scope of change.  I'll put up a PR for bumping to latest ES after I can sort out what's gone wrong with building the image at all.

File: cluster/addons/fluentd-elasticsearch/es-image/elasticsearch_logging_discovery.go (hunk 1 above)
Comment (coffeepac) [reply cluster/addons/fluentd-elasticsearch/es-image/elasticsearch_logging_discovery.go]: I had to pin to a much later version of kubernetes (this is old code) and then do the module replace dance.  should pass the linting/verify CI checks now.

**Review by monotek — APPROVED** [review]

**Comment by coffeepac** [comment]
@monotek it needs the `/lgtm` if you think its good to go :)

**Comment by monotek** [comment]
/lgtm

Damn, sorry 😁

**Comment by coffeepac** [comment]
thanks @monotek !

**Pull request merged**